tests/test_global.py

```diff
@@ -0,0 +1,292 @@
+        "E:/data/test/20230513_104139_PNEO-03_1_1_30cm_RD_12bit_RGBNED_NoordwijkAanZee_Coepelduynen_cropped_Test_region_cropped_re_ndvi.tif",
+        ["ndwi"],
+    )
+
+    # Open the TIFF image and display ndvi
+    with rasterio.open(filepath_ndwi) as src:
+        # Get the number of bands
+        num_bands = src.count
+        print("Number of bands:", num_bands)
+
+        assert num_bands >= 8, "ndwi bands not added correctly"
+
+    os.remove(filepath_ndwi)
+
+
+filepath = ""
+
+
+def test_adding_ndvi_superview():
+    filepath = georegion.execute_link(
+        "https://api.satellietdataportaal.nl/v1/download/SV_RD_11bit_RGBI_50cm/20191202_110523_SV1-04",
+        add_ndvi_band=True,
+        plot=False,
+    )
+
+    # Open the TIFF image and display ndvi
+    with rasterio.open(filepath) as src:
+        # Get the number of bands
+        num_bands = src.count
+        print("Number of bands:", num_bands)
+
+        assert num_bands >= 5, "ndvi bands not added correctly"
+
+
+def test_adding_ndwi_superview():
+    # Add the NDWI Channel
+    filepath_ndwi = nso_manipulator.add_index_channels(
+        "E:/data/test/20191202_110523_SV1-04_50cm_RD_11bit_RGBI_NoordwijkAanZee_Coepelduynen_cropped_Test_region_cropped_ndvi.tif",
+        ["ndwi"],
+    )
+
+    # Open the TIFF image and display ndvi
+    with rasterio.open(filepath_ndwi) as src:
+        # Get the number of bands
+        num_bands = src.count
+        print("Number of bands:", num_bands)
+
+        assert num_bands >= 6, "re_ndvi bands not added correctly"
+
+    os.remove(filepath_ndwi)
+
+
+# Test Cloud detection
+def test_cloud_detection():
+
+    path_to_model = pickle.load(
+        open(
+            "C:/repos/satellite-images-nso/models/cloud_detection_logistic_regression_v1.0.sav",
+            "rb",
+        )
+    )
+    cloud_pictures = [
+        "E:/data/coepelduynen/20220514_114854_SV1-02_SV_RD_11bit_RGBI_50cm_Rijnsburg_natura2000_coepelduynen_cropped.tif",
+        "E:/data/coepelduynen/20190302_105829_SV1-01_50cm_RD_11bit_RGBI_KatwijkAanZee_natura2000_coepelduynen_cropped.tif",
+        "E:/data/coepelduynen/20191202_110525_SV1-04_50cm_RD_11bit_RGBI_KatwijkAanZee_natura2000_coepelduynen_cropped.tif",
+    ]
+    non_cloud_pictures = [
+        "E:/data/coepelduynen/20230402_105321_PNEO-03_1_49_30cm_RD_12bit_RGBNED_Zoeterwoude_natura2000_coepelduynen_cropped.tif",
+        "E:/data/coepelduynen/20230513_104139_PNEO-03_1_1_30cm_RD_12bit_RGBNED_NoordwijkAanZee_natura2000_coepelduynen_cropped.tif",
+        "E:/data/coepelduynen/20220515_113347_SV1-02_SV_RD_11bit_RGBI_50cm_KatwijkAanZee_natura2000_coepelduynen_cropped.tif",
+    ]
+
+    positive_pictures = [
+        detect_clouds(model=path_to_model, filepath=afile) for afile in cloud_pictures
+    ]
+    false_pictures = [
+        detect_clouds(model=path_to_model, filepath=afile)
+        for afile in non_cloud_pictures
+    ]
+
+    assert all_true(positive_pictures) and all_false(
```

Comment:
You know what the results should be right? Which ones are False and which ones True?, why this weird way of checking?

Reply:
This is unanswered.

Reply:
i don't really understand this, this line checks if all the positive cloud images are indeed positive and the false ones are indeed false. Easy way to check in one line

Reply:
Ah, you're right. I must have somehow misread this. Apologies.